include/alpaka/mem/buf/Traits.hpp

```diff
@@ -175,9 +176,10 @@ namespace alpaka
         TPlatform const& platform,
         TExtent const& extent = TExtent())
     {
-        if constexpr(hasMappedBufSupport<TPlatform>)
+        using Platform = Platform<TPlatform>;
```

Comment:
CI error: 

```
/home/runner/work/alpaka/alpaka/include/alpaka/mem/buf/Traits.hpp:179:26: error: 'Platform' does not name a type; did you mean 'TPlatform'?
  179 |         using Platform = Platform<TPlatform>;
      |                          ^~~~~~~~
      |                          TPlatform
```

The include to `alpaka::platform/Traits.hpp` is missing.

Reply:
Won't `Platform = Platform<TPlatform` also trigger some kind of shadowing warning?

Reply:
I've changes the name to `PlatformForMappedAlloc` (which is used nowhere else currently).
Is this okay?

Reply:
Yes, that's fine.